app/src/main/kotlin/com/hippo/ehviewer/ui/main/DownloadCard.kt

```diff
@@ -173,40 +184,76 @@ fun DownloadCard(
-                    val icon = remember {
-                        movableContentOf<Boolean> {
-                            Icon(imageVector = if (it) Icons.Default.Pause else Icons.Default.PlayArrow, contentDescription = null)
-                        }
-                    }
-                    if (selectMode) {
-                        Box(modifier = Modifier.minimumInteractiveComponentSize()) {
-                            icon(running)
+                    viewToMeasure = {
+                        Text(text = "", style = MaterialTheme.typography.labelLarge)
+                    },
+                ) { measuredHeight ->
+                    Row(
+                        verticalAlignment = Alignment.CenterVertically,
+                        modifier = Modifier.height(measuredHeight * 2 + 2.dp).offset(x = 8.dp)
+                    ) {
+                        val running =
+                            downloadState == DownloadInfo.STATE_WAIT || downloadState == DownloadInfo.STATE_DOWNLOAD
+                        val icon = remember {
+                            movableContentOf<Boolean> {
+                                Icon(
+                                    imageVector = if (it) Icons.Default.Pause else Icons.Default.PlayArrow,
+                                    contentDescription = null,
+                                    modifier = Modifier.size(measuredHeight * 1.25f),
+                                )
+                            }
                         }
-                    } else {
-                        IconButton(onClick = if (running) onStop else onStart) {
-                            icon(running)
+                        if (selectMode) {
+                            Box(modifier = Modifier.minimumInteractiveComponentSize()) {
+                                icon(running)
+                            }
+                        } else {
+                            IconButton(onClick = if (running) onStop else onStart) {
+                                icon(running)
+                            }
                         }
                     }
                 }
+
                 if (stateText != null) {
                     Text(
                         text = stateText,
                         modifier = Modifier.constrainAs(stateTextRef) {
-                            end.linkTo(parent.end, 16.dp)
+                            end.linkTo(parent.end)
                             bottom.linkTo(actionsRef.top)
                         },
-                        style = MaterialTheme.typography.titleSmall,
+                        style = MaterialTheme.typography.labelLarge,
                     )
                 }
             }
         }
     }
 }
+
+// https://stackoverflow.com/questions/70500071/is-it-possible-to-measure-string-width-to-properly-size-a-text-composable
+@Composable
+internal fun MeasureUnconstrainedViewHeight(
```

Comment:
I would like to avoid this
Can't we just calculate the height(dp) by `fontScale` and `labelLarge` fontSize?

Reply:
@mkx173 
https://github.com/revonateB0T/EhViewer/commit/b15337bd531778f692691b09a01c76b3b0f325f2
Does it work as intended for you?

Reply:
Also see https://github.com/revonateB0T/EhViewer/commit/4fffa045166a78a952f92484026cb5a3e6aa9cb0

Reply:
Yes this seems to work well here. I cherry picked your commits to this pull request.